indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -939,10 +939,19 @@ public Response killPendingSegments(
     }
 
     if (taskMaster.isLeader()) {
-      final int numDeleted = indexerMetadataStorageAdapter.deletePendingSegments(dataSource, deleteInterval);
-      return Response.ok().entity(ImmutableMap.of("numDeleted", numDeleted)).build();
+      try {
+        final int numDeleted = indexerMetadataStorageAdapter.deletePendingSegments(dataSource, deleteInterval);
+        return Response.ok().entity(ImmutableMap.of("numDeleted", numDeleted)).build();
+      }
+      catch (DruidException e) {
+        return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
The code should be `400` (bad request) only if the `IndexerMetadataStorageAdapter` threw an invalid input exception. There might be other exceptions in which case we should not give a 400 response.

I don't recall exactly but I think there was a way to convert a `DruidException` to a `Response` and it would automatically pick up the right HTTP status code.

Reply:
Extracted the code from `e.getStatusCode()`, which uses the underlying exception category's status code. Also, added a generic catch block to handle non-DruidException and updated the tests